crates/proto/src/clients/mod.rs

```diff
@@ -0,0 +1,396 @@
+//! gRPC client builder utilities for Miden node.
+//!
+//! This module provides a unified type-state [`Builder`] for creating various gRPC clients with
+//! explicit configuration decisions for TLS, timeout, and metadata.
```

Comment:
nit: should "type-state" be "type-safe"?

Reply:
I think `type-state` is correct in this context, the state is typed :)